pyccel/codegen/printing/ccode.py

```diff
@@ -972,6 +974,16 @@ def _print_Assign(self, expr):
             return '{};'.format(self._print(expr.rhs))
         lhs = self._print(expr.lhs)
         rhs = expr.rhs
+        if isinstance(rhs, IndexedElement) and isinstance(expr.lhs, Variable):
+            free_code = ''
+            for i in rhs.args:
+                if isinstance(i, Slice):
+                    free_code += 'free_array(%s);\n' % self._print(expr.lhs.name)
```

Comment:
Am I right in saying that this handles:
```
a = b[1,:]
```
by printing
```
free_array(a);
a = ...
```
?
Is the `free_array` always necessary? This looks like it is a similar case to the `Allocate` node where a `status` is required

Reply:
yes, this handles assigning to an already allocated variable (`status` can be unknown) but I don't have access to the Allocate node in this case i don't think it's necessary to add the if condition all around the code we can add it implicitly to`free_array`

the `if` condition is just to optimize so we don't need to call all the free inside the function